CRM/Core/Resources.php

```diff
@@ -781,11 +781,15 @@ public function coreResourceList($region) {
    *   is this page request an ajax snippet?
    */
   public static function isAjaxMode() {
-    return in_array(CRM_Utils_Array::value('snippet', $_REQUEST), array(
+    if (in_array(CRM_Utils_Array::value('snippet', $_REQUEST), array(
         CRM_Core_Smarty::PRINT_SNIPPET,
         CRM_Core_Smarty::PRINT_NOFORM,
         CRM_Core_Smarty::PRINT_JSON,
-      ));
+      ))
+    ) {
+      return TRUE;
+    }
+    return strpos(CRM_Utils_System::getUrlPath(), 'civicrm/ajax') === 0;
```

Comment:
I can definitely confirm there are a lot of ajax paths & checking for example civicrm/ajax/checkphone it is being called from the template without snippet=1 AND it is calling an ajax function - so it should be treated as ajax mode.

I guess there could be an edge case risk that someone put ajax in a url that is not an ajax page. I spot checked a handful & didn't find any